meerkat_libs/db_adapters/dynamodb.py

```diff
@@ -0,0 +1,100 @@
+import boto3
+import logging
+
+
+class DynamoDBAdapter():
+
+    def __init__(self, db_url, structure):
+        self.structure = structure
+        self.db_url = db_url
+        self.keys = self._extract_keys()
```

Comment:
I think that key extraction should happen during the setup. It seems like a little too much for the constructor.
Moreover it depends on the self.structure to be initialised which is odd.

I suggest moving the call to `self._extract_keys()` into the `setup()` method.

Reply:
I think I disagree. This function is not about doing any actual setup of the DB.  There is some information buried deep in the provided structure - the keys.  This function simply factors out the process of accessing that information. It get's the info once on creation and stores it in a much more convienient form to be accessed by any of the DB interaction methods. i.e. it is about defining "how to interact" rather than actually "interacting" with the db which is why I think it should be in the constructor.

Reply:
After giving it a second thought I think you're right. Also I've just learned that '__init__` method is not the constructor but indeed an initialiser. https://stackoverflow.com/a/938677